demos/sample_azure_iot/sample_azure_iot.c

```diff
@@ -0,0 +1,625 @@
+ * @param[out] ppucIothubDeviceId  Pointer to uint8_t* deviceId return from Provisioning Service
+ * @param[in,out] pulIothubDeviceIdLength  Length of deviceId
+ */
+    static uint32_t prvIoTHubInfoGet( NetworkCredentials_t * pXNetworkCredentials,
+                                      uint8_t ** ppucIothubHostname,
+                                      uint32_t * pulIothubHostnameLength,
+                                      uint8_t ** ppucIothubDeviceId,
+                                      uint32_t * pulIothubDeviceIdLength );
+
+#endif /* democonfigENABLE_DPS_SAMPLE */
+
+/**
+ * @brief The task used to demonstrate the MQTT API.
+ *
+ * @param[in] pvParameters Parameters as passed at the time of task creation. Not
+ * used in this example.
+ */
+static void prvAzureDemoTask( void * pvParameters );
+
+
+/**
+ * @brief Connect to MQTT broker with reconnection retries.
+ *
+ * If connection fails, retry is attempted after a timeout.
+ * Timeout value will exponentially increase until maximum
+ * timeout value is reached or the number of attempts are exhausted.
+ *
+ * @param[out] pxNetworkContext The parameter to return the created network context.
+ *
+ * @return The status of the final connection attempt.
+ */
+static TlsTransportStatus_t prvConnectToServerWithBackoffRetries( const char * pcHostName,
+                                                                  uint32_t ulPort,
+                                                                  NetworkCredentials_t * pxNetworkCredentials,
+                                                                  NetworkContext_t * pxNetworkContext );
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Static buffer used to hold MQTT messages being sent and received.
+ */
+static uint8_t ucSharedBuffer[ democonfigNETWORK_BUFFER_SIZE ];
+
+/**
+ * @brief Global start unix time
+ */
+static uint64_t ulGlobalEntryTime = 1639093301;
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Cloud message callback handler
+ */
+static void prvHandleCloudMessage( AzureIoTHubClientCloudToDeviceMessageRequest_t * pxMessage,
+                                   void * pvContext )
+{
+    ( void ) pvContext;
+
+    LogInfo( ( "Cloud message payload : %.*s \r\n",
+               pxMessage->ulPayloadLength,
+               pxMessage->pvMessagePayload ) );
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Direct method message callback handler
+ */
+static void prvHandleDirectMethod( AzureIoTHubClientMethodRequest_t * pxMessage,
+                                   void * pvContext )
+{
+    LogInfo( ( "Method payload : %.*s \r\n",
+               pxMessage->ulPayloadLength,
+               pxMessage->pvMessagePayload ) );
+
+    AzureIoTHubClient_t * xHandle = ( AzureIoTHubClient_t * ) pvContext;
+
+    if( AzureIoTHubClient_SendMethodResponse( xHandle, pxMessage, 200,
```

Comment:
PART1 - 
@danewalton / @hihigupt  - I just realized `AzureIoTHubClient_SendMethodResponse` in product takes `AzureIoTHubClientMethodRequest_t` as an input parameter.

Not this PR I know, but i think this should be a MethodResponse?  We might add response specific fields at some point, and even today the componentName/commandName I think would be ignored on response.

PART2 - 
It looks like this is echoing back whatever server sent to us?  That seems fair enough, no matter what we do with the MethodResponse i suggest in part1.  But we should clearly comment that that's the intent here.

Reply:
PART1 : I agree we can revisit if we need the response struct.

PART2: I did not do that because in that case I have to construct JSON response that contains random payload we got in method. To keep it simple I just return empty json.

Reply:
added details in issue: for now resolving this.